src/components/Inputs/SelectWithDropdown/SelectWithDropdown.tsx

```diff
@@ -0,0 +1,69 @@
+import { SetStateAction, useEffect, useRef, useState } from "react";
+import style from "./SelectWithDropdown.module.scss";
+import ArrowDown from "@/components/Buttons/ArrowButtons/ArrowDown";
+import ArrowUp from "@/components/Buttons/ArrowButtons/ArrowUp";
+import useOutsideClick from "@/services/hooks/useOutsideClick";
+
+interface ISelectDropdown {
+  data: string[];
+  registerId: number;
+  registerName: string;
+  register: (...args: any[]) => any;
```

Comment:
**nit:** same here. Try to not pass `register`. Focus on native HTML element attribute names instead of library related names. Remember that `{...register}` syntax returns native input props. You don't need to pass it to the component - just make possible to pass specific props